src/analysis/history.cpp

```diff
@@ -279,4 +280,100 @@ void ReduceLocalizationFunction(MeshData<Real> *md) {
 
 } // exp
 
+using namespace parthenon::package::prelude;
+template <typename F>
+KOKKOS_INLINE_FUNCTION Real ComputeDivInPillbox(int ndim, int b, int k, int j, int i,
+                                                const parthenon::Coordinates_t &coords,
+                                                const F &f) {
+  Real div_mass_flux_integral;
+  div_mass_flux_integral =
+      -(f(b, 1, k, j, i + 1) - f(b, 1, k, j, i)) * coords.FaceArea<X1DIR>(k, j, i);
+
+  if (ndim >= 2) {
+    div_mass_flux_integral +=
+        -(f(b, 2, k, j + 1, i) - f(b, 2, k, j, i)) * coords.FaceArea<X2DIR>(k, j, i);
+  }
+  if (ndim >= 3) {
+    div_mass_flux_integral +=
+        -(f(b, 3, k + 1, j, i) - f(b, 3, k, j, i)) * coords.FaceArea<X3DIR>(k, j, i);
+  }
+  return div_mass_flux_integral;
+}
```

Comment:
Maybe move this into `history.hpp` so it can be re-used.

Reply:
done